internal/eval/eval.go

```diff
@@ -661,6 +657,79 @@ func evalJSONGET(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(string(resultBytes), false)
 }
 
+// evalJSONTOGGLE toggles a boolean value stored at the specified key and path.
+// args must contain at least the key and path (where the boolean is located).
+// If the key does not exist or is expired, it returns response.RespNIL.
+// If the field at the specified path is not a boolean, it returns an encoded error response.
+// If the boolean is `true`, it toggles to `false` (returns :0), and if `false`, it toggles to `true` (returns :1).
+// Returns an encoded error response if the incorrect number of arguments is provided.
+func evalJSONTOGGLE(args []string, store *dstore.Store) []byte {
+    if len(args) < 2 {
+        return diceerrors.NewErrArity("JSON.TOGGLE")
+    }
+
+    key := args[0]
+    path := args[1]
+
+    // Retrieve the object from the database
+    obj := store.Get(key)
+    if obj == nil {
+        return []byte("-ERR could not perform this operation on a key that doesn't exist\r\n")
+    }
+
+    // Check if the object is of JSON type
+    errWithMessage := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeJSON, object.ObjEncodingJSON)
+    if errWithMessage != nil {
+        return errWithMessage
+    }
+
+    jsonData := obj.Value
+
+    // Parse the JSONPath expression
+    expr, err := jp.ParseString(path)
+    if err != nil {
+        return diceerrors.NewErrWithMessage("invalid JSONPath")
+    }
+
+    // Execute the JSONPath query
+    results := expr.Get(jsonData)
+    if len(results) == 0 {
+        return clientio.Encode([]interface{}{}, false)  // Return empty array if no matches
+    }
+
+	toggleResults := make([]interface{}, len(results))
+
+    for i, result := range results {
+        switch v := result.(type) {
+        case bool:
+            newValue := !v
+            err := expr.Set(jsonData, newValue)
```

Comment:
Hey, I am implementing JSON.ARRAPPEND where I also have to modify json for multiple results like this. I had also done this similarly using expr.Set() in the loop, it works fine when there is only 1 value in results array but for example in your case lets say the json is `{"isSimple": true,"nested":{"isSimple":false}}` and you use the toggle command with path $..isSimple does it make both of them same or does it toggle both of them separately?
I am facing this issue where its setting the same array which i am getting during the last iteration and expr.Set sets it to all the matching paths not just the one in iteration.

Reply:
Hey @srivastava-yash , I face the same issue. I am working on similar command json.numincrby.